independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java

```diff
@@ -159,7 +159,20 @@ public CuratedApplication bootstrap() throws BootstrapException {
                 appModelFactory.setEnableClasspathCache(true);
             }
         }
-        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig);
+        CurationResult curationResult = appModelFactory.resolveAppModel();
+        if (curationResult.getApplicationModel().getApplicationModule() != null
```

Comment:
`curationResult.getApplicationModel().getAppArtifact()` would be simpler and a bit more reliable here.

Reply:
right, I pushed a fix.